### PR TITLE
Makes dchat controlled humans resist while moving

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Title, deadchat controlled humans will now try a standard resist when they move if they're grabbed.

## Why It's Good For The Game
Makes it a bit harder for deadchat controlled mobs to be stopped in their tracks with a single grab.

## Testing
Grabbed a human, deadchat controlled them, and tried to move them.

## Changelog
:cl:
tweak: Deadchat controlled humans will now try to resist out of grabs when moving.
/:cl:

### CODE REVIEW

**Comment by SynthTwo** [comment]
I will miss zooming at mach speed as a dchat controlled mob

**Comment by SpringSkipper** [comment]
> I will miss zooming at mach speed as a dchat controlled mob

this doesn't change that though?

**Comment by lewcc** [comment]
> > I will miss zooming at mach speed as a dchat controlled mob
> 
> this doesn't change that though?

Yeah, if anything it should make movement easier

File: code/modules/mob/living/carbon/human/human_mob.dm [hunk 1]
```diff
@@ -2200,6 +2200,22 @@ Eyes need to have significantly high darksight to shine unless the mob has the X
...
+/datum/component/deadchat_control/human/Initialize(_deadchat_mode, _inputs, _input_cooldown, _on_removal)
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	_inputs["up"] = CALLBACK(parent, TYPE_PROC_REF(/mob/living/carbon/human, dchat_step), NORTH)
+	_inputs["down"] = CALLBACK(parent, TYPE_PROC_REF(/mob/living/carbon/human, dchat_step), SOUTH)
+	_inputs["left"] = CALLBACK(parent, TYPE_PROC_REF(/mob/living/carbon/human, dchat_step), WEST)
+	_inputs["right"] = CALLBACK(parent, TYPE_PROC_REF(/mob/living/carbon/human, dchat_step), EAST)
+
+	return ..()
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/human/human_mob.dm:2217]: this should be in `deadchat_control.dm`. adding it to a 2000+ line file is less than ideal.

**Review by hal9000PR — DISMISSED** [review]

File: code/modules/mob/living/carbon/human/human_mob.dm (hunk 1 above)
Comment (lewcc) [reply code/modules/mob/living/carbon/human/human_mob.dm]: Ended up moving the new stuff out to a new file, and moving this new datum into `deadchat_control.dm`

**Review by S34NW — APPROVED** [review]

**Pull request merged**